lib/husky_musher/app.py

```diff
@@ -127,21 +135,41 @@ def main():
         # If not in REDCap project, create new record
         user_data = {'record_id': register_net_id(net_id)}
 
-    # TODO -- generate a survey link for a particular day
-    # We are awaiting finalization of the REDCap project to know how
-    # daily attestations (repeating instruments) will be implemented.
+    # Because of REDCap's survey queue logic, we can point a participant to an
+    # upstream survey. If they've completed it, REDCap will automatically direct
+    # them to the next, uncompleted survey in the queue.
+    event = 'enrollment_arm_1'
+    instrument = 'eligibility_screening'
+    repeat_instance = None
+
+    # If all enrollment event instruments are complete, point participants
+    # to today's daily attestation instrument.
+    # If the participant has already completed the daily attestation,
+    # REDCap will prevent the participant from filling out the survey again.
     if is_complete('eligibility_screening', user_data) and \
         is_complete('consent_form', user_data) and \
         is_complete('enrollment_questionnaire', user_data):
-        return f"Congrats, {net_id}, you're already registered under record ID " \
-            f"{user_data['record_id']} and your eligibility " \
-            "screening, consent form, and enrollment questionnaires are complete!"
 
-    # Generate a link to the eligibility questionnaire, and then redirect.
-    # Because of REDCap's survey queue logic, we can point a participant to an
-    # upstream survey. If they've completed it, REDCap will automatically direct
-    # them to the next, uncompleted survey in the queue.
-    return redirect(generate_survey_link(user_data['record_id']))
+        event = 'encounter_arm_1'
+        instrument = 'daily_attestation'
+        # Repeat instance number should be days since the start of the study,
+        # with the first instance starting at 1.
+        repeat_instance = (datetime.today() - STUDY_START_DATE).days
+
+        if repeat_instance < 0:
+            # This should never happen!
+            app.logger.error("Failed to create a valid repeat instance")
+            return ERROR_MESSAGE
+
+        if repeat_instance == 0:
+            ### TODO: Update with new language about daily attestations start.
+            attestation_start = (STUDY_START_DATE + timedelta(days=1)).strftime("%B %d, %Y")
+            return ("Your enrollment is complete!\n" +
+                    f"Daily attestations start on {attestation_start}.\n" +
+                    "You will receive a daily reminder to complete your attestations.")
+
+    # Generate a link to the appropriate questionnaire, and then redirect.
+    return redirect(generate_survey_link(user_data['record_id'], event, instrument, repeat_instance))
```

Comment:
We may want some error handling here in case the survey link doesn't generate correctly for some reason.

Reply:
Thanks for the suggestion! I wrapped all of our interactions with the REDCap API in try blocks in the latest commit.